### ISSUE TITLE
handle paranemic crossovers in cadnano export

### ISSUE THREAD

**Issue by dave-doty** [UC-Davis-molecular-computing/scadnano-python-package#185]
scadnano allows "paranemic" crossovers (https://doi.org/10.1021/ja038381e, https://doi.org/10.1021/acs.chemrev.8b00207), i.e., crossovers that connect two domains in the same direction:

![image](https://user-images.githubusercontent.com/19274365/127682924-453318f8-0686-4d8f-8a44-e759601cdde5.png)

Exporting to cadnano (Export --> cadnano v2 from menu) and then reimporting that same `.json` file (File --> Import cadnano v2 from menu) gives an error:

![image](https://user-images.githubusercontent.com/19274365/127683015-fa94dadc-62b1-4407-9d6c-5c45026cb35f.png)

The first domain mentioned in the error exists, and is the 3'-most domain of the scaffold. But the second domain (helix=2, start=15, end=16) does not exist in the original design.

The scadnano file with this design is

```json
{
  "version": "0.16.1",
  "grid": "square",
  "helices": [
    {"grid_position": [0, 0], "max_offset": 17},
    {"grid_position": [0, 1], "max_offset": 17},
    {"grid_position": [0, 2], "max_offset": 17}
  ],
  "strands": [
    {
      "color": "#0066cc",
      "is_scaffold": true,
      "domains": [
        {"helix": 0, "forward": true, "start": 0, "end": 8},
        {"helix": 2, "forward": true, "start": 7, "end": 16}
      ]
    }
  ]
}
```

Note that because the above strand is a scaffold, and we want the strand to be forward on both helices, we must have both helices be even due to cadnano limitations on directions scaffold/staple strands can go on helices based on their parity.

However, the same error happens if the strand is a staple, and is reversed on both helices:

![image](https://user-images.githubusercontent.com/19274365/127683354-58cdb858-da28-472e-908b-997e64171cd6.png)

**Comment by tcosmo** [comment]
Paranemic crossovers are possible in cadnano thus bug must be internal to conversion algorithm.

I added `test_inputs/test_paranemic_crossover.json`, the import from cadnano v2 works, the export fails.

![Screenshot 2021-08-02 at 14 01 00](https://user-images.githubusercontent.com/7616128/127866106-77cc2c33-4ab9-4d1d-a159-2318ebdf68b1.png)

**Comment by tcosmo** [comment]
I believe that I fixed the bug, there was a subtle index setting error in the export code

**Comment by dave-doty** [comment]
Closed in #189.
